connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/operate/OperateClientAdapterImpl.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.connector.runtime.inbound.operate;
+
+import io.camunda.connector.runtime.core.inbound.OperateClientAdapter;
+import io.camunda.connector.runtime.inbound.importer.ProcessDefinitionSearch;
+import io.camunda.operate.dto.FlownodeInstance;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class OperateClientAdapterImpl implements OperateClientAdapter {
+
+  private final ProcessDefinitionSearch processDefinitionSearch;
+
+  public OperateClientAdapterImpl(final ProcessDefinitionSearch processDefinitionSearch) {
+    this.processDefinitionSearch = processDefinitionSearch;
+  }
```

Comment:
Are you aware that the process definition search is stateful? I would recommend to not use it in this context.

Reply:
Please just use the operate client here.